File: id3/java/src/id3/Id3.java
package id3;

import trainningSet.TrainningSet;

public class Id3 {
	
	Node root;
	TrainningSet set;
	
	/**
	 * This class performs the Id3 mining proccess
	 */
	public Node mine(Node n, TrainningSet set) {
		if (set.hasOnlyOneValue()) {
			return set.getMoreFrequentValue(); 
		}
		Node root = set.getBestEntropy();
		for (Node b: root.getBranches()) {
			mine(b, set.subSet(b));
		}
		return root;
	}

}

File: id3/python/id3.py
"""."""


class ID3:
    """description."""

    def __init__(self):
        """."""
        self.root = None
        self.trainning_set = None

    def mine(self, node, trainning_set):
        """."""
        root = trainning_set.getBestEntropy()
        if trainning_set.hasOnlyOneValue:
            return trainning_set.getMoreFrequentValue()
        else:
            for n in root.getBranches():
                self.mine(n, trainning_set.subSet(n))
        return root
